lib/routes/account.js

```diff
@@ -974,6 +976,33 @@ module.exports = (log, db, mailer, Password, config, customs, checkPassword, pus
           }
         }
 
+        function sendVerifyLoginPush(devices) {
+          if (doSigninConfirmation) {
+            log.info({
+              op: 'account.signin.confirm.push.start',
+              uid: accountRecord.uid,
+              tokenVerificationId: tokenVerificationId
+            })
+
+            return push.notifyVerifyLogin(accountRecord.uid, devices)
+          }
+        }
+
+        function sendVerifyLogin() {
+          return db.devices(accountRecord.uid).then(
+            function (devices) {
+              devices = devices.filter(function(device) {
+                return (device.type === 'desktop' && parseFloat(device.uaBrowserVersion) >= 56)
+              })
+              if (devices.length === 0 && !(features.isPushLoginVerifyEnabled(accountRecord.uid, accountRecord.email))) {
```

Comment:
We should probably move the feature gate check to the top of this function, that way, if it isn't enabled, we can skip the database read. So, something like:

```js
if (features.isPushLoginVerifyEnabled(accountRecord.uid, accountRecord.email)) {
  return db.devices().then( ... )
} else {
  return sendVerifyLoginEmail()
}
```